crates/api_models/src/payments.rs

```diff
@@ -4969,6 +5012,68 @@ pub struct GooglePaySessionResponse {
     pub secrets: Option<SecretInfoToInitiateSdk>,
 }
 
+#[derive(Debug, Clone, Eq, PartialEq, serde::Serialize, ToSchema)]
+#[serde(rename_all = "lowercase")]
+pub struct SamsungPaySessionTokenResponse {
+    /// Samsung Pay API version
+    pub version: String,
+    /// Samsung Pay service ID to which session call needs to be made
+    pub service_id: String,
+    /// Order number of the transaction
+    pub order_number: String,
+    /// Filed containing merchant information
+    #[serde(rename = "merchant")]
+    pub merchant_payment_information: MerchantPaymentInformation,
+    /// Filed containing the payment amount
+    pub amount: AmountDetails,
+    /// Payment protocol type
+    pub protocol: ProtocolType,
```

Comment:
Same for `MerchantPaymentInformation`, `AmountDetails` and `AmountFormat`, aren't they going to be Samsung Pay specific?